### PR TITLE
added vegan adonis for multi-way permanova tests

### LINKED ISSUE #123: multi-way PERMANOVA
PERMANOVA and Adonis currently only support one-way group comparisons in `q2-diversity`.

However, the [vegan-R versions](http://cc.oulu.fi/~jarioksa/softhelp/vegan/html/adonis.html) accept functions as input to support multi-way comparisons (e.g., beta diversity differences as a function of an individual's age, BMI, zodiac sign). 

That would be **awesome** to support in `q2-diversity`.

@mortonjt 's comments in [this q2-sample-classifer` issue](https://github.com/nbokulich/q2-sample-classifier/issues/30) may prove useful here, re: how to input a formula to test interactions between variables.

### PR DESCRIPTION
**Pull request by nbokulich**
fixes #123 

# Questions
1. I added `r-vegan` to `ci/recipe/meta.yaml`. Is that all we need to do to add that dependency?

### CODE REVIEW

**Comment by thermokarst** [comment]
Yep, that is how you add a new dep! If you want to add that dep in a separate PR, we can let busywork build a new env file w the dep, then travis should be able to run on this PR.

**Comment by nbokulich** [comment]
thanks! will cut a new PR in a moment.

travis is failing for other reasons — things I did not touch in this PR. Should I fix those too?

File: q2_diversity/_beta/adonis_assets/run_adonis.R [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+suppressWarnings(library(vegan))
+
+
+### LOAD DATA ###
+
+distances <- read.table(file = args[[1]], sep="\t", header=TRUE, fill=TRUE, row.names=1)
+sample.md <- read.table(file = args[[2]], sep="\t", header=TRUE, fill=TRUE, row.names=1)
+formula <- args[[3]]
+perms <- as.integer(args[[4]])
+njobs <- args[[5]]
```
[the remark this answers is not in the document]
Comment (nbokulich) [reply q2_diversity/_beta/adonis_assets/run_adonis.R]: no I have tested... evidently it is interpreted as a string by R?

[the remark this answers is not in the document]
Comment (ebolyen) [reply q2_diversity/_beta/adonis_assets/run_adonis.R]: yeah, that's not too surprising I suppose

File: q2_diversity/plugin_setup.py [hunk 2]
```diff
@@ -711,3 +711,37 @@
...
         citations['spearman1904proof']]
 )
+
+plugin.visualizers.register_function(
+    function=q2_diversity.adonis,
+    inputs={'distance_matrix': DistanceMatrix},
+    parameters={'metadata': Metadata,
+                'formula': Str,
+                'permutations': Int % Range(1, None)
+                'n_jobs': Int % Range(0, None)},
```
Comment (ebolyen) [inline q2_diversity/plugin_setup.py]: Sorry, messed this up the first time when I suggested:
```suggestion
                'n_jobs': Int % Range(1, None)},
```

**Pull request merged**